mono/mini/mini-runtime.c

```diff
@@ -2597,8 +2597,24 @@ mono_llvmonly_fallback_imt_tramp (gpointer *arg, MonoMethod *imt_method)
 {
 	int i = 0;
 
-	while (arg [i] && arg [i] != imt_method)
-		i += 2;
+	while (arg [i]) {
+		MonoMethod *candidate = arg [i];
+		gboolean same = (arg [i] == imt_method);
+
+		if (!same && imt_method->klass->is_array_special_interface && candidate->klass->is_array_special_interface) {
+			MonoClass *gklass = mono_class_get_generic_class (imt_method->klass)->container_class;
```

Comment:
This check looks wrong, it's too geral.

Can't we use the same code we have in the JIT virtual call trampoline to figure out the same information?

There's a lot of nuance here that we need to take care.

- the array special interface checks only applies if the source type is an array.
- it should not match all instances of the same method. For example, methods in IList<int> should not match the IList<double> from double[].